datafusion/expr/src/built_in_function.rs

```diff
@@ -1162,8 +1163,9 @@ impl BuiltinScalarFunction {
                 Signature::uniform(2, vec![Int64], self.volatility())
             }
             BuiltinScalarFunction::ArrowTypeof => Signature::any(1, self.volatility()),
-            BuiltinScalarFunction::Abs
-            | BuiltinScalarFunction::Acos
+            BuiltinScalarFunction::Abs => Signature::any(1, self.volatility()),
```

Comment:
Defining a signature for the Decimal type seems impossible.

Reply:
@alamb 
In the function signature, I can’t find a way to declare the parameter type as a generic decimal type that supports arbitrary precision and scale. Just like this :
```rust
BuiltinScalarFunction::Abs => {
       // Compilation failed
       Signature::uniform(1, vec![Decimal128(_, _)], self.volatility())
}
```
If a function requires a decimal type as its parameter, we have to declare it as an inexact `Any` type.